draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -9628,15 +9630,85 @@ Content-Encoding: gzip
-   matches the target URI's authority component
-   (see <xref target="RFC2818"/>). Correctly implementing such verification
-   can be difficult (see <xref target="Georgiev"/>).
+   The "https" scheme (<xref target="https.uri"/>) is intended to prevent (or
+   at least reveal) many of these potential attacks on establishing
+   authority, provided that the negotiated TLS connection is secured and the
+   client properly verifies that the communicating server's identity matches
+   the target URI's authority component, as described below. Correctly
+   implementing such verification can be difficult (see <xref
+   target="Georgiev"/>).
+</t>
+<section title="Identifying HTTPS Servers" anchor="https.identify.server">
+<t>
+   In general, HTTP/TLS requests are generated by dereferencing a URI.
+   As a consequence, the hostname for the server is known to the client.
+   If the hostname is available, the client &MUST; check it against the
+   server's identity as presented in the server's Certificate message,
+   in order to prevent man-in-the-middle attacks.
+</t>
+<t>
+   If the client has external information as to the expected identity of
+   the server, the hostname check &MAY; be omitted. (For instance, a
+   client may be connecting to a machine whose address and hostname are
+   dynamic but the client knows the certificate that the server will
+   present.) In such cases, it is important to narrow the scope of
+   acceptable certificates as much as possible in order to prevent man
+   in the middle attacks.  In special cases, it may be appropriate for
+   the client to simply ignore the server's identity, but it must be
+   understood that this leaves the connection open to active attack.
+</t>
+<t>
+   If a subjectAltName extension of type dNSName is present, that &MUST;
+   be used as the identity. Otherwise, the (most specific) Common Name
+   field in the Subject field of the certificate &MUST; be used. Although
+   the use of the Common Name is existing practice, it is deprecated and
+   Certification Authorities are encouraged to use the dNSName instead.
+</t>
+<t>
+   Matching is performed using the matching rules specified by
+   <xref target="RFC5280"/>.  If more than one identity of a given type is present in
+   the certificate (e.g., more than one dNSName name, a match in any one
+   of the set is considered acceptable.) Names may contain the wildcard
+   character * which is considered to match any single domain name
+   component or component fragment. E.g., *.a.com matches foo.a.com but
+   not bar.foo.a.com. f*.com matches foo.com but not bar.com.
+</t>
+<t>
+   In some cases, the URI is specified as an IP address rather than a
+   hostname. In this case, the iPAddress subjectAltName must be present
+   in the certificate and must exactly match the IP in the URI.
+</t>
+<t>
+   If the hostname does not match the identity in the certificate, user
+   oriented clients MUST either notify the user (clients MAY give the
```

Comment:
`user-oriented`, no?

Reply:
is a "user(-)oriented client" different from a "user agent"?